### ISSUE TITLE
Implement GradientDescentOptimizer

### ISSUE THREAD

**Issue by migueldeicaza** [migueldeicaza/TensorFlowSharp#98]
In Python Tensorflow, that is a highlevel operation that would be convenient to have in C#:

python/training/gradient_descent.py

**Comment by mdabros** [comment]
I have started to work on this with the goal of being able to finish the LinearRegression sample from the SampleTest project. 

If anybody has already started to implement the GradientDecentOptimizer, or want to contribute, let me know. The branch I am working on can be found here:
https://github.com/mdabros/TensorFlowSharp/tree/gradient-descent-optimizer

**Comment by MaximG1234** [comment]
Would be great to have this

**Comment by gmlwns2000** [comment]
+1

**Comment by alexpantyukhin** [comment]
@mdabros I think it would be okay if you make a WIP PR if you already has some results.

**Comment by mdabros** [comment]
Sadly I have not been able to spent much time on this, so the work is still very preliminary. Basically only the definition of some of the base types. Most of the work can be found here in the training namespace:
[Training](https://github.com/mdabros/TensorFlowSharp/tree/gradient-descent-optimizer/TensorFlowSharp/Training).

To be realistic I probably won't have time to continue this work until after December, so if anybody wants to contribute or take over, feel free to copy anything useful from my branch.

**Comment by alexpantyukhin** [comment]
@mdabros I think the most sutable variant of you commit what you already have as WIP pull request (for keeping your work as yours). After that someone will continue.

**Comment by mdabros** [comment]
@alexpantyukhin Sure, if this is the preferred approach, I will submit a WIP pull request as soon as possible.

**Comment by praeclarum** [comment]
Just popping in to say I would like this too. Tensorflow bindings are a bit useless if you can't train networks.

**Comment by ZivKidd** [comment]
looking forward for this

**Comment by lurium** [comment]
Yes, this would be great.
